third_party/sl4a/src/main/java/com/google/android/mobly/snippet/rpc/MethodDescriptor.java

```diff
@@ -267,25 +272,99 @@ String getHelp() {
      */
     public static Object getDefaultValue(Type parameterType, Annotation[] annotations) {
         for (Annotation a : annotations) {
-            if (a instanceof RpcOptional) {
+            if (a instanceof RpcDefault) {
+                RpcDefault defaultAnnotation = (RpcDefault) a;
+                TypeConverter<?> converter = converterFor(parameterType, defaultAnnotation.converter());
+                return converter.convert(defaultAnnotation.value());
+            } else if (a instanceof RpcOptional) {
                 return null;
             }
         }
         throw new IllegalStateException("No default value for " + parameterType);
     }
 
+    @SuppressWarnings("rawtypes")
+    private static TypeConverter<?> converterFor(
+            Type parameterType, Class<? extends TypeConverter> converterClass) {
+        if (converterClass == TypeConverter.class) {
+            TypeConverter<?> converter = typeConverters.get(parameterType);
+            if (converter == null) {
+                throw new IllegalArgumentException(
+                        "No predefined converter found for " + parameterType);
+            }
+            return converter;
+        }
+        try {
+            Constructor<?> constructor = converterClass.getConstructor(new Class<?>[0]);
+            return (TypeConverter<?>) constructor.newInstance(new Object[0]);
+        } catch (Exception e) {
+            throw new IllegalArgumentException(
+                    "Cannot create converter from " + converterClass.getCanonicalName());
+        }
+    }
+
     /**
      * Determines whether or not this parameter has default value.
      *
      * @param annotations annotations of the parameter
      */
     public static boolean hasDefaultValue(Annotation[] annotations) {
         for (Annotation a : annotations) {
-            if (a instanceof RpcOptional) {
+            if (a instanceof RpcDefault || a instanceof RpcOptional) {
                 return true;
             }
         }
         return false;
     }
 
+    /**
+     * Returns the converters for {@code String}, {@code Integer}, {@code Long},
+     * and {@code Boolean}.
+     */
+    private static Map<Class<?>, TypeConverter<?>> populateConverters() {
+        Map<Class<?>, TypeConverter<?>> converters = new HashMap<Class<?>, TypeConverter<?>>();
+        converters.put(String.class, new TypeConverter<String>() {
+            @Override
+            public String convert(String value) {
+                return value;
+            }
+        });
+        converters.put(Integer.class, new TypeConverter<Integer>() {
+            @Override
+            public Integer convert(String input) {
+                try {
+                    return Integer.decode(input);
+                } catch (NumberFormatException e) {
+                    throw new IllegalArgumentException("'" + input + "' is not an integer");
```

Comment:
can we use string format instead of `+` which can make code formatting look better?

Reply:
Changed to `String.format`.